src/sqswatcher/plugins/slurm.py

```diff
@@ -100,16 +171,32 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
             node_name = "NodeName={0}".format(event.host.hostname)
             node_list = [node for node in node_list if node.split()[0] != node_name]
         elif event.action == "ADD":
-            # Add new node
-            new_node = "NodeName={nodename} CPUs={cpus} State=UNKNOWN\n".format(
-                nodename=event.host.hostname, cpus=event.host.slots
-            )
+            # Add new node and gres_node depending on whether the instance has GPU
+            if "gpus" in instance_properties:
+                new_node = "NodeName={nodename} CPUs={cpus} Gres=gpu:tesla:{gpus} State=UNKNOWN\n".format(
+                    nodename=event.host.hostname, cpus=event.host.slots, gpus=instance_properties["gpus"]
```

Comment:
I'm wondering if we could send the gpu info in the event as we are doing for slots for the following reasons:
- in case of an update of the instance type we are sure we are using the correct value even if the instance properties hasn't updated yet. dummy-nodes are recomputed on updates of instance_properties but real nodes aren't
- it is slightly easier to move to a multi-instance type model